agent/config/builder.go

```diff
@@ -867,9 +867,6 @@ func (b *Builder) checkVal(v *CheckDefinition) *structs.CheckDefinition {
 	}
 
 	id := types.CheckID(b.stringVal(v.ID))
-	if v.CheckID != nil {
-		id = types.CheckID(b.stringVal(v.CheckID))
-	}
```

Comment:
I see it got removed here but there isn't an entry for id->checkID in TranslateKeys above?